toolkit/offchain/src/collateral_selection.rs

```diff
@@ -0,0 +1,74 @@
+use ogmios_client::types::*;
+
+pub const MIN_REQUIRED_COLLATERAL: u64 = 5_000_000;
+
+// Largest-First selection algorithm based on https://cips.cardano.org/cip/CIP-0002#largest-first
+pub fn largest_first(
+	inputs_available: &[OgmiosUtxo],
+	amount_required: u64,
+	max_collateral_inputs: u32,
+	min_required_collateral: u64,
+) -> Result<Vec<&OgmiosUtxo>, &str> {
+	let amount_required = amount_required.max(min_required_collateral);
+	let amount_available: u64 = inputs_available.iter().map(|utxo| utxo.value.lovelace).sum();
+	if amount_available < amount_required {
```

Comment:
Is this a trade-off / optimization to not sort inputs if there is no chance of finding the collateral input?

Reply:
The check has to happen anyways and I wasn't sure if Rust is allowed to reorder the sorting and the check so it seemed better to write this way. (I did not check with Godbolt, yet.)